docs/tools/near-indexer.md

```diff
@@ -0,0 +1,487 @@
+            sync_mode: near_indexer::SyncModeEnum::FromInterruption,
+            await_for_node_synced: near_indexer::AwaitForNodeSyncedEnum::WaitForFullSync,
+            };
+    },
+    _ => panic!("ERROR: You have to pass `init` or `run` arg."),
+}
+
+```
+
+- If you like, you can refactor the code a bit by creating a `command` variable leaving your final `main()` function looking something like this:
+
+```rs
+fn main() {
+    let args: Vec<String> = std::env::args().collect();
+    let home_dir = std::path::PathBuf::from(near_indexer::get_default_home());
+
+    let command = args.get(1)
+        .map(|arg| arg.as_str())
+        .expect("You need to provide a command: `init` or `run` as arg");
+
+    match command {
+        "init" => near_indexer::init_configs(
+        &home_dir,
+        Some("localnet"),
+        None,
+        None,
+        1,
+        false,
+        None,
+        true,
+        None,
+    ),
+        "run" => {
+            let indexer_config = near_indexer::IndexerConfig {
+              home_dir: std::path::PathBuf::from(near_indexer::get_default_home()),
+              sync_mode: near_indexer::SyncModeEnum::FromInterruption,
+              await_for_node_synced: near_indexer::AwaitForNodeSyncedEnum::WaitForFullSync,
+              };
+
+    actix::System::builder()
+        .stop_on_panic(true)
+        .run(move || {
+            let indexer = near_indexer::Indexer::new(indexer_config);
+            let stream = indexer.streamer();
+            actix::spawn(listen_blocks(stream));
+        })
+        .unwrap();
+        }
+        _ => panic!("You have to pass `init` or `run` arg"),
+    };
+}
+```
+
+### Initialize configurations
+
+- Last step in in configuring your network is to initialize. To do this, simply run:
+
+```bash
+cargo run --init
+```
+
+## Run the Indexer
+
+- You're ready to start the Indexer! To run enter the following command in your terminal:
+
+```bash
+cargo run -- run
+```
+
+- Ouch! You most likely got the following error:
+
+```bash
+thread 'main' panicked at 'Indexer should track at least one shard. 
+Tip: You may want to update /Users/joshford/.near/config.json with `"tracked_shards": [0]`
+            ', /Users/joshford/.cargo/git/checkouts/nearcore-5bf7818cf2261fd0/d7b0ca4/chain/indexer/src/lib.rs:65:9
+note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
+Panic in Arbiter thread, shutting down system.
+```
+
+- Don't worry, this one is an easy fix!
+
+  - Open your `config.json` located in the `.near` folder in the root of your home directory. _( ~/.near/config.json )_
+  - In this file, locate: `"tracked_shards": []` and change the value to [0].
+  - Save the file and try running your indexer again.
+
+```bash
+cargo run -- run
+```
+
+It’s alive! 🎉 
+
+You should now see print to your terminal of `StreamerMessage` structures like the example below. :)
+
+<details>
+<summary>**Example Stream:**</summary>
+<p>
+
+```json
+StreamerMessage {
```

Comment:
@amgando suggested to make make it a JSON object and give a few examples of what you can do with JSON using `jq` and I'm gonna try this, so it would be changed.

Reply:
Ahh yes, I was going to ask about adding a section that would explain how users can return different results. :)